src/quickfix.c

```diff
@@ -2480,9 +2720,13 @@ copy_loclist_stack(win_T *from, win_T *to)
     if (qi == NULL)		    // no location list to copy
 	return;
 
-    // allocate a new location list
-    if ((to->w_llist = qf_alloc_stack(QFLT_LOCATION)) == NULL)
+    // allocate a new location list, set size of stack to 'from' window value
+    if ((to->w_llist = qf_alloc_stack(QFLT_LOCATION,
+		    from->w_p_lhi)) == NULL)
```

Comment:
This can be on one line.

```suggestion
    if ((to->w_llist = qf_alloc_stack(QFLT_LOCATION, from->w_p_lhi)) == NULL)
```